tests-e2e/core/gitopsDeployment_test.go

```diff
@@ -0,0 +1,534 @@
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+				),
+			)
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+
+		})
+
+		It("Should ensure synchronicity of create and update of GitOpsDeployment", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("ensurng no updates are done in existing deployment, if CR is submitted again without any changes")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				repoURL, "environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			gitOpsDeploymentResource.Spec.Source.RepoURL = repoURL
+			gitOpsDeploymentResource.Spec.Source.Path = "environments/overlays/dev"
+
+			err = k8s.Update(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&gitOpsDeploymentResource), &gitOpsDeploymentResource)
+			Expect(gitOpsDeploymentResource.Spec.Source.RepoURL).To(Equal(repoURL))
+			Expect(gitOpsDeploymentResource.Spec.Source.Path).To(Equal("environments/overlays/dev"))
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+				),
+			)
+
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+		})
+
+		It("Should ensure synchronicity of create and update of GitOpsDeployment", func() {
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+			By("ensuring GitOpsDeployment should update successfully on changing value(s) within Spec")
+
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource(name,
+				"https://github.com/redhat-appstudio/managed-gitops", "manifests/appstudio-controller-rbac",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			gitOpsDeploymentResource.Spec.Source.RepoURL = repoURL
+			gitOpsDeploymentResource.Spec.Source.Path = "environments/overlays/dev"
+
+			err = k8s.Update(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&gitOpsDeploymentResource), &gitOpsDeploymentResource)
```

Comment:
Likewise, after `Update`, and before we our `Synced`/`Healthy`, we should wait for (`Eventually`) a resource from the Git repository to be deployed. This resource should be one that is ONLY in the new repository URL, so that we can be sure that an update has taken place. See parent review comment for some further details on this.